### ISSUE TITLE
test_arange failure - Jetson TX2 (CPU)

### ISSUE THREAD

**Issue by KellenSunderland** [apache/incubator-mxnet#8383]
## Description
The test test_operator.test_init is consistently failing.

## Environment info (Required)
Jetson TX2, aarch64, Ubuntu 16.04, Jetpack 3.1 full install.

```
What to do:
1. Flash your Jetson TX2 with Jetpack 3.1, full install. 
2. Modify make/config.mk to indicate use of Openblas.
3. Compile MXNet locally and install.
4. Run nosetests --verbose tests/python/unittest
```

Package used (Python/R/Scala/Julia):
Python package, compiled from source.

## Build info (Required if built from source)

Compiler (gcc/clang/mingw/visual studio):
gcc 

MXNet commit hash:
050d30452eb7421045bacbc29d27622a96d8aff9

Build config:
(Paste the content of config.mk, or the build command.)
Diff with commit:
```bash
-ifeq ($(UNAME_S), Darwin)
-USE_BLAS = apple
-else
-USE_BLAS = atlas
-endif
+USE_BLAS = openblas
```
## Error Message:
======================================================================
FAIL: test_operator.test_init
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/nvidia/incubator-mxnet-cpu/tests/python/unittest/test_operator.py", line 2569, in test_init
    test_arange()
  File "/home/nvidia/incubator-mxnet-cpu/tests/python/unittest/test_operator.py", line 2564, in test_arange
    assert_almost_equal(np_out, nd_out.asnumpy())
  File "/home/nvidia/incubator-mxnet-cpu/python/mxnet/test_utils.py", line 490, in assert_almost_equal
    raise AssertionError(msg)
AssertionError:
Items are not equal:
Error 508000.000000 exceeds tolerance rtol=0.000010, atol=0.000000.  Location of maximum error:(1,), a=48.000000, b=50.000000
 a: array([ 50,  48,  46, ..., 212, 210, 208], dtype=uint8)
 b: array([50, 50, 50, ..., 50, 50, 50], dtype=uint8)

**Comment by KellenSunderland** [comment]
Minimal reproducible code:

```python
import mxnet as mx
import numpy as np
import random

from mxnet.test_utils import *

dtype = [np.uint8]
config = (50, -50, -2)

repeats = random.choice([1,3])
np_out = np.repeat(np.arange(*config, dtype=dtype), repeats)
nd_out = mx.nd.arange(*config, repeat=repeats, dtype=dtype)
assert_almost_equal(np_out, nd_out.asnumpy())
```

**Comment by KellenSunderland** [comment]
I'm not sure if this is a valid test.  It think the desired behaviour is somewhat undefined given that we're trying to create a range with an endpoint of -50 on an unsigned data structure.  It could be possible to at the least duplicate numpy's behaviour here, which seems to be to overflow without throwing an error.

**Comment by KellenSunderland** [comment]
Here's a print of the relevant output on an OSX build

```
In [3]: print(np_out)
[ 50  48  46  44  42  40  38  36  34  32  30  28  26  24  22  20  18  16
  14  12  10   8   6   4   2   0 254 252 250 248 246 244 242 240 238 236
 234 232 230 228 226 224 222 220 218 216 214 212 210 208]

In [4]: print(nd_out)

[ 50  48  46  44  42  40  38  36  34  32  30  28  26  24  22  20  18  16
  14  12  10   8   6   4   2   0 254 252 250 248 246 244 242 240 238 236
 234 232 230 228 226 224 222 220 218 216 214 212 210 208]
<NDArray 50 @cpu(0)>
```

**Comment by szha** [comment]
@apache/mxnet-committers: This issue has been inactive for the past 90 days. It has no label and needs triage.

For general "how-to" questions, our [user forum](https://discuss.mxnet.io/) (and [Chinese version](https://discuss.gluon.ai/)) is a good place to get help.

**Comment by haojin2** [comment]
@KellenSunderland Since numpy has a defined behavior for such inputs I think the test is valid. I also tried this on ubuntu and mac, and both produced same results as numpy. Maybe this is a platform-specific thing? Seems like here: https://github.com/apache/incubator-mxnet/blob/master/src/operator/tensor/init_op.h#L449-L452 already has some code to address the signed-unsigned issue. Seems like there's some problem with the step conversion as the each output entry from MXNet arange has the same value 50 which is the start value.

**Comment by KellenSunderland** [comment]
Yeah sorry I thought this issue was closed.  I believe it's been fixed.  Give me some time and I'll verify.

**Comment by haojin2** [comment]
@KellenSunderland Thanks for your quick response! Please close the issue if the results on your end shows that this has been solved.

**Comment by KellenSunderland** [comment]
Sorry was traveling. Will try to take a look at this one today.

**Comment by KellenSunderland** [comment]
Verified this was fixed prior to 1.2.  Just did a test on-device.  Closing ...
